frankenphp.c

```diff
@@ -105,13 +106,23 @@ static uintptr_t frankenphp_clean_server_context() {
 
 static void frankenphp_request_reset() {
   zend_try {
+    frankenphp_server_context *ctx = SG(server_context);
+
     int i;
+    if (ctx->worker_ready) {
+      // Restore worker script super globals
+      for (i = 0; i < NUM_TRACK_VARS; i++) {
+        zval_ptr_dtor(&PG(http_globals)[i]);
+      }
+      memcpy(&PG(http_globals), &ctx->worker_http_globals,
+             sizeof(zval) * NUM_TRACK_VARS);
 
-    for (i = 0; i < NUM_TRACK_VARS; i++) {
-      zval_ptr_dtor(&PG(http_globals)[i]);
+      php_hash_environment();
```

Comment:
For this to work, we need to save and restore the `SG(request_info)` of the worker script, but I wonder if it is worth it. Resetting the whole context as in the beginning of the script will be costly and useless. After all, maybe should we keep the current behavior and keep the context of the previous request until the next one is handled.

WDYT @withinboredom @TimWolla?

Reply:
As I'm mentioned in the issue I created, I don’t have an opinion here. That depends on what you believe your users expectations are.

Reply:
That's a tricky one. Personally, I use the last context to run logging things (treblle specifically) before continuing to the next request so I would find it useful. I can't think of why you would not want the last request's context.